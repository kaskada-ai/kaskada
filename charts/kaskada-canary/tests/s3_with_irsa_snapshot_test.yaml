suite: s3 with irsa snapshot test
templates:
  - deployment.yaml
  - persistantVolumeClaims.yaml
  - service.yaml
  - serviceAccount.yaml
release:
  name: my-release
  namespace: my-namespace
  revision: 1
  upgrade: true
values: 
  - ./values/s3_with_irsa.yaml
tests:
  - it: manifest should match snapshot
    asserts:
      - matchSnapshot: {}
