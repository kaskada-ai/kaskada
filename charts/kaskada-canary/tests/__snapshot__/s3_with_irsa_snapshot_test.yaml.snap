manifest should match snapshot:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app.kubernetes.io/instance: my-release
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: kaskada-canary
        helm.sh/chart: kaskada-canary-0.0.1
      name: my-release-kaskada-canary
      namespace: s3-with-irsa
    spec:
      replicas: 1
      selector:
        matchLabels:
          app.kubernetes.io/instance: my-release
          app.kubernetes.io/name: kaskada-canary
      strategy:
        type: Recreate
      template:
        metadata:
          labels:
            app.kubernetes.io/instance: my-release
            app.kubernetes.io/name: kaskada-canary
        spec:
          containers:
            - command:
                - /bin/sparrow-main
                - serve
              env:
                - name: SPARROW_LOG_FILTER
                  value: egg::=warn,sparrow_=info,info
                - name: SPARROW_LOG_JSON
                  value: "true"
              image: ghcr.io/kaskada-ai/kaskada/engine:latest
              imagePullPolicy: IfNotPresent
              name: engine
              ports:
                - containerPort: 50052
                  name: grpc
                  protocol: TCP
              volumeMounts:
                - mountPath: /tmp
                  name: kaskada-canary-tmp
            - command:
                - /bin/wren
              env:
                - name: DB_DIALECT
                  value: sqlite
                - name: DB_PATH
                  value: /db/kaskada.db
                - name: DB_IN_MEMORY
                  value: "false"
                - name: OBJECT_STORE_BUCKET
                  value: kaskada-us-west-2-demo
                - name: OBJECT_STORE_PATH
                  value: /data/
                - name: OBJECT_STORE_TYPE
                  value: s3
                - name: DEBUG
                  value: "false"
                - name: LOG_FORMAT_JSON
                  value: "true"
              image: ghcr.io/kaskada-ai/kaskada/engine:latest
              imagePullPolicy: IfNotPresent
              name: manager
              ports:
                - containerPort: 50051
                  name: grpc
                  protocol: TCP
                - containerPort: 3365
                  name: rest
                  protocol: TCP
              volumeMounts:
                - mountPath: /db
                  name: kaskada-canary-db
          serviceAccountName: kaskada
          volumes:
            - name: kaskada-canary-db
              persistentVolumeClaim:
                claimName: kaskada-canary-db
            - ephemeral:
                volumeClaimTemplate:
                  metadata:
                    labels:
                      type: kaskada-canary-tmp
                  spec:
                    accessModes:
                      - ReadWriteOnce
                    resources:
                      requests:
                        storage: 10Gi
                    storageClassName: gp2
              name: kaskada-canary-tmp
  2: |
    apiVersion: v1
    kind: PersistentVolumeClaim
    metadata:
      name: kaskada-canary-db
      namespace: s3-with-irsa
    spec:
      accessModes:
        - ReadWriteOnce
      resources:
        requests:
          storage: 1Gi
      storageClassName: gp2
      volumeMode: Filesystem
  3: |
    apiVersion: v1
    kind: Service
    metadata:
      labels:
        app.kubernetes.io/instance: my-release
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: kaskada-canary
        helm.sh/chart: kaskada-canary-0.0.1
      name: my-release-kaskada-canary
      namespace: s3-with-irsa
    spec:
      ports:
        - name: grpc
          port: 50051
          protocol: TCP
          targetPort: 50051
        - name: rest
          port: 3365
          protocol: TCP
          targetPort: 3365
      selector:
        app.kubernetes.io/instance: my-release
        app.kubernetes.io/name: kaskada-canary
      type: LoadBalancer
  4: |
    apiVersion: v1
    kind: ServiceAccount
    metadata:
      annotations:
        eks.amazonaws.com/role-arn: arn:aws:iam::581634802795:role/irsa_s3
      labels:
        app.kubernetes.io/instance: my-release
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: kaskada-canary
        helm.sh/chart: kaskada-canary-0.0.1
      name: kaskada
      namespace: s3-with-irsa
